src/KM_Controls.pas

```diff
@@ -142,6 +137,11 @@   TKMControl = class
     function GetSelfHeight: Integer; virtual;
     function GetSelfWidth: Integer; virtual;
     procedure UpdateVisibility; virtual;
+    procedure ControlMouseDown(Sender: TObject; Shift: TShiftState); virtual;
+    procedure ControlMouseUp(Sender: TObject; Shift: TShiftState); virtual;
+    procedure FocusChanged(aFocused: Boolean); virtual;

```

Comment:
Added new protected procedure FocusChanged. Invoked when control focus status is chaned.
Used by any other Control (f.e. Memo/Edit) when they need to do some procedures on change focus.

Before that I used external OnFocus event for that purposes - which is supposed to be used by external users (outside of TKMControls unit). 
So there was a bug, when outside of TKMEdit you could overrid OnFocus event and then this edit will stop unselect text on unfocus.

By using FocusChanged procedure for internal controls 'magic' and OnFocusChange event for external users we fix this kind of possible bugs.